src/lightning/fabric/fabric.py

```diff
@@ -266,7 +269,7 @@ def setup(
 
         if optimizers:
             # join both types in a tuple for API convenience
-            return (module, *optimizers)
+            return (module, *optimizers, scheduler)
```

Comment:
This is a breaking change, it will cause existing user code to fail, because `scheduler` is returned unconditionally.

Since `scheduler` is `Optional` in the signature, I suggest we only return it if it was not `None` as an argument, so we won't break anyone's code.